File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/security/config/JWTConfig.java
package com.ntnu.gidd.security.config;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@Configuration
public class JWTConfig {

    @Value("${security.jwt.uri:/auth}")
    private String uri;

    @Value("${security.jwt.header:Authorization}")
    private String header;

    @Value("${security.jwt.prefix:Bearer }")
    private String prefix;

    @Value("${security.jwt.expiration:#{24*60*60}}")
    private int expiration;

    @Value("${security.jwt.secret:JwtSecretKey}")
    private String secret;

    @Value("${security.jwt.refresh_expiration:#{1000*60*60*24}}")
    private int refreshExpiration;

}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/security/config/JWTConfig.kt
package ntnu.idatt2105.security.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration

@Configuration
data class JWTConfig(
    @Value("\${security.jwt.uri:/auth}")
    var uri: String,

    @Value("\${security.jwt.header:Authorization}")
    var header: String,

    @Value("\${security.jwt.prefix:Bearer }")
    var prefix: String,

    @Value("\${security.jwt.expiration:#{24*60*60}}")
    var expiration: Int,

    @Value("\${security.jwt.secret:JwtSecretKey}")
    var secret: String,

    @Value("\${security.jwt.refresh_expiration:#{1000*60*60*24}}")
    var refreshExpiration: Int,
)
